build/charts/theia/templates/clickhouse/clickhouseinstallation.yaml

```diff
@@ -16,8 +16,20 @@ spec:
       {{ .Values.clickhouse.connectionSecret.readOnlyUsername }}/networks/ip: "::/0"
     profiles:
       readonly/readonly: 1
+    {{- if .Values.clickhouse.service.secureConnection.enable }}
+    files:
+      {{- include "clickhouse.tlsConfig" (dict "Files" .Files "Global" .) | indent 6 }}
+    {{- end }}
     clusters:
       - name: "clickhouse"
+        {{- if .Values.clickhouse.service.secureConnection.enable }}
+        secure: "yes"
+        settings:
+          tcp_port: {{ .Values.clickhouse.service.tcpPort }} # keep for localhost
```

Comment:
Do we want to expose tcp_port and http_port when secureConnection is disabled?
I only enclose https_port and tcp_port_secure inisde the double curly braket.
I'm not quite familiar with the settings, are they optional?

Reply:
Thanks for pointing out this! I've moved it out.

Let me explain about how it works.

The port here defines for the port in ClickHouse pod while the port exposed by the service defined in the Service template: https://github.com/antrea-io/theia/blob/aaeab26bb8c368ab0f243ded087148d73d280e41/build/charts/theia/templates/clickhouse/clickhouseinstallation.yaml#L66-L70

Although we may prefer to get consistency between these two, ClickHouse itself has some internal calls on the tcp port 9000 before the ClickHouse server is ready for the ClickHouse service, i.e. 9000 is hard-coded in ClickHouse code and I do not find a way to overwrite them easily. Thus finally my solution is to make the internal tcp port always 9000 while others will be consistent with the customization. And this will not affect the user experience.